src/Common/Menus/Traits/With_Settings.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Provides methods and properties for submenus.
+ *
+ * @since   TBD
+ *
+ * @package TEC\Common\Menus
+ */
+
+namespace TEC\Common\Menus\Traits;
+
+trait With_Settings {
```

Comment:
On the same note as the Admin Bar one, if this one will be used only by menu or sub-menu pages, then I would just include it there.